src/spec/Schema.php

```diff
@@ -124,6 +124,10 @@ protected function attributeDefaults(): array
             'allOf' => null,
             'oneOf' => null,
             'anyOf' => null,
+            // nullable is only relevant, when a type is specified
+            // return null as default when there is no type
+            // return false as default when there is a type
+            'nullable' => $this->hasProperty('type') ? false : null,
```

Comment:
At what point this is calculated? Is `schema->type` mutable? Is this possible?
```php
        // nullable is undefined if no type is given
        $schema = Reader::readFromJson('{"oneOf": [{"type": "string"}, {"type": "integer"}]}', Schema::class);
        $this->assertNull($schema->type);
        $this->assertNull($schema->nullable);
        $schema->type = Type::STRING;
        $this->assertFalse($schema->nullable);
```

If it's not mutable - looks good

Reply:
this is executed on demand, e.g. in case the property is accessed.

Reply:
ok, then I think we need to fix this as a testcase